// react-cartify/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const Cart = () => {
  const [items, setItems] = React.useState([]);
  const [itemName, setItemName] = React.useState('');
  const [itemPrice, setItemPrice] = React.useState('');
  const addItem = () => {
    setItems([...items, {
      name: itemName,
      price: parseFloat(itemPrice)
    }]);
    setItemName('');
    setItemPrice('');
  };
  const removeItem = index => {
    setItems(items.filter((_, i) => i !== index));
  };
  const totalPrice = items.reduce((total, item) => total + item.price, 0);
  return /*#__PURE__*/React__default["default"].createElement("div", null, /*#__PURE__*/React__default["default"].createElement("h2", null, "Shopping Cart"), /*#__PURE__*/React__default["default"].createElement("input", {
    type: "text",
    value: itemName,
    onChange: e => setItemName(e.target.value),
    placeholder: "Item Name"
  }), /*#__PURE__*/React__default["default"].createElement("input", {
    type: "number",
    value: itemPrice,
    onChange: e => setItemPrice(e.target.value),
    placeholder: "Item Price"
  }), /*#__PURE__*/React__default["default"].createElement("button", {
    onClick: addItem
  }, "Add Item"), /*#__PURE__*/React__default["default"].createElement("ul", null, items.map((item, index) => /*#__PURE__*/React__default["default"].createElement("li", {
    key: index
  }, item.name, " - $", item.price.toFixed(2), ' ', /*#__PURE__*/React__default["default"].createElement("button", {
    onClick: () => removeItem(index)
  }, "Remove")))), /*#__PURE__*/React__default["default"].createElement("h3", null, "Total: $", totalPrice.toFixed(2)));
};

exports.Cart = Cart;
